receiver/vcenterreceiver/metadata.yaml

```diff
@@ -99,12 +108,98 @@ attributes:
       - "on"
       - "off"
       - "suspended"
+  entity_status:
+    name_override: status
+    description: The current status of the managed entity.
+    type: string
+    enum:
+      - "red"
+      - "yellow"
+      - "green"
+      - "gray"
   object_name:
     name_override: object
     description: The object on the virtual machine or host that is being reported on.
     type: string
 
 metrics:
+  vcenter.datacenter.cluster.count:
+    enabled: false
+    description: The number of clusters under the datacenter.
+    unit: "{clusters}"
+    sum:
+      monotonic: false
+      value_type: int
+      aggregation_temporality: cumulative
+    attributes: [entity_status]
+    warnings:
+      if_enabled_not_set: "this metric will be enabled by default starting in release v0.106.0"
+  vcenter.datacenter.host.count:
+    enabled: false
+    description: The number of hosts under the datacenter.
+    unit: "{hosts}"
+    sum:
+      monotonic: false
+      value_type: int
+      aggregation_temporality: cumulative
+    attributes: [entity_status, host_power_state]
+    warnings:
+      if_enabled_not_set: "this metric will be enabled by default starting in release v0.106.0"
+  vcenter.datacenter.vm.count:
+    enabled: false
+    description: The number of VM's under the datacenter.
+    unit: "{virtual_machines}"
+    sum:
+      monotonic: false
+      value_type: int
+      aggregation_temporality: cumulative
+    attributes: [entity_status, vm_count_power_state]
+    warnings:
+      if_enabled_not_set: "this metric will be enabled by default starting in release v0.106.0"
+  vcenter.datacenter.datastore.count:
+    enabled: false
+    description: The number of datastores under the datacenter.
+    unit: "{datastores}"
+    sum:
+      monotonic: false
+      value_type: int
+      aggregation_temporality: cumulative
+    attributes: []
+    warnings:
+      if_enabled_not_set: "this metric will be enabled by default starting in release v0.106.0"
+  vcenter.datacenter.disk.space:
```

Comment:
disk.total or disk.limit instead?

Reply:
So the equivalent datastore metric is "disk.usage" with a "disk_state" of `available` and `used` which I don't think is necessarily the best.

`disk.space` is better than `disk.usage` at the least i think for this info. what do you think about `disk.size` or `disk.capacity`? I'm not sure about any of these to be honest.

`disk_state` seems like a bit of an odd name for the attribute here. what about `disk_space_type` if you stick with `disk_space`?

Reply:
I went with disk.space since it implies an absolute metric, whereas usage can sometimes be used on metrics with percentages. Although I am not opposed to capacity, it seems vSphere instills some confusions around when capacity actually comes into use and what it actually means. I wanted to avoid this confusion if possible.

As for disk_state, this was already an attribute that existed before this PR that relayed the concept of having disk space being used/available so I used it. I am not opposed to changing the naming of this attribute, but if we do, we would have to do it in another PR as its used for another metric.

Reply:
resolve